synth/src/datasource/mysql_datasource.rs

```diff
@@ -0,0 +1,309 @@
+use sqlx::{Pool, MySql, Row, Column, TypeInfo};
+use anyhow::{Result, Context};
+use crate::datasource::DataSource;
+use async_std::task;
+use sqlx::mysql::{MySqlPoolOptions, MySqlQueryResult, MySqlRow, MySqlColumn};
+use serde_json::{Value, Map, Number};
+use async_trait::async_trait;
+use crate::datasource::relational_datasource::{RelationalDataSource, ColumnInfo, PrimaryKey, ForeignKey, ValueWrapper};
+use std::prelude::rust_2015::Result::Ok;
+use std::convert::TryFrom;
+use synth_core::Content;
+use rust_decimal::Decimal;
+use rust_decimal::prelude::ToPrimitive;
+use synth_core::schema::{StringContent, RegexContent, NumberContent, RangeStep, DateTimeContent, ChronoValueType};
+use synth_core::schema::number_content::{I64, F64, U64};
+
+/// TODO
+/// Known issues:
+/// - MySql aliases bool and boolean data types as tinyint. We currently define all tinyint as i8.
+///   Ideally, the user can define a way to force certain fields as bool rather than i8.
+
+pub struct MySqlDataSource {
+    pool: Pool<MySql>,
+    connect_params: String
+}
+
+#[async_trait]
+impl DataSource for MySqlDataSource {
+    type ConnectParams = String;
+
+    fn new(connect_params: &Self::ConnectParams) -> Result<Self> {
+        task::block_on(async {
+            let pool = MySqlPoolOptions::new()
+                .max_connections(3) //TODO expose this as a user config?
+                .connect(connect_params.as_str())
+                .await?;
+
+            Ok::<Self, anyhow::Error>(MySqlDataSource {
+                pool,
+                connect_params: connect_params.to_string()
+            })
+        })
+    }
+
+    async fn insert_data(&self, collection_name: String, collection: &[Value]) -> Result<()> {
+        self.insert_relational_data(collection_name, collection).await.unwrap();
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl RelationalDataSource for MySqlDataSource {
+    type QueryResult = MySqlQueryResult;
+
+    async fn execute_query(&self, query: String, _query_params: Vec<&str>) -> Result<MySqlQueryResult> {
+        let result = sqlx::query(query.as_str())
+            .execute(&self.pool)
+            .await?;
+
+        Ok(result)
+    }
+
+    fn get_catalog(&self) -> Result<&str> {
+        self.connect_params
```

Comment:
Same as the postgres comment here. Perhaps we can use `ConnectOptions` to parse the DB name for us instead of doing string-fu :)